sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -1220,7 +1236,24 @@ private[client] class Shim_v2_1 extends Shim_v2_0 {
 
 private[client] class Shim_v2_2 extends Shim_v2_1
 
-private[client] class Shim_v2_3 extends Shim_v2_1
+private[client] class Shim_v2_3 extends Shim_v2_1 {
+  private lazy val getTablesByTypeMethod =
+    findMethod(
+      classOf[Hive],
+      "getTablesByType",
+      classOf[String],
+      classOf[String],
+      classOf[TableType])
```

Comment:
BTW, did you test this API on JDK11?
Like SPARK-29245, the remote invocation should be tested manually.

Reply:
I see, I'll test it manually.

Reply:
@dongjoon-hyun @maropu Would you plz share some experience on building with hive nightly version `2.3.7-SNAPSHOT`. I tried with modify hive version in `pom.xml`, but it didn't work.